web3/auto/ipc.py

```diff
@@ -0,0 +1,28 @@
+from os import path
+
+from web3 import (
+    IPCProvider,
+    Web3
+)
+
+
+def ipc():
+
+    home = path.expanduser('~')
+
+    # The following 2 lines are default paths
+    parity_path = home + '/.local/share/io.parity.ethereum/jsonrpc.ipc'
+    geth_path = home + '/.ethereum/geth.ipc'
```

Comment:
It doesn't look like this will cover the major operating systems, for geth. Take a look at how the IPC module does it: https://github.com/pipermerriam/web3.py/blob/master/web3/providers/ipc.py#L38-L65

In fact, it might be better to just initialize `IPCProvider()` without any args, which will automatically look for the geth file.

---

Nice that it looks for the parity path! Can you add in support for parity on other OS's, too?

Reply:
I would be supportive of just adding the parity paths to the `web3.providers.ipc.get_default_ipc_path` function.  In fact, I think it would be good to modify that function so that rather than just returning a path, it enumerates a number of different paths and checks to see if there is an IPC socket in place that it can connect to and returns the first one.

Order of operations I would like to see on that is:

1. go-ethereum (mainnet)
2. go-ethereum (rinkeby)
3. go-ethereum (ropsten)
4. parity (mainnet)
5. parity (ropsten)
6. parity (kovan)